app/code/Magento/Store/Model/App/Emulation.php

```diff
@@ -183,6 +183,16 @@ public function stopEnvironmentEmulation()
         return $this;
     }
 
+    /**
+     * Checks whether the environment is being emulated
+     *
+     * @return bool
+     */
+    public function isEnvironmentEmulated()
```

Comment:
Hi @dnsv. Thanks for the collaboration, due to [Magento backward-compatible guide](http://devdocs.magento.com/guides/v2.2/contributor-guide/backward-compatible-development/) and [Magento Technical Guidelines](https://devdocs.magento.com/guides/v2.2/coding-standards/technical-guidelines.html) we can create new public methods only in special cases where there are really needed. In the current case, please make this function `private`

Reply:
Hi @VladimirZaets. `isEnvironmentEmulated()` is called in `\Magento\Theme\Model\Theme` and `\Magento\Theme\Model\View\Design`. The method therefore needs to be public.